src/ray/object_manager/plasma/store.cc

```diff
@@ -623,9 +623,16 @@ int PlasmaStore::RemoveFromClientObjectIds(const ObjectID &object_id,
 }
 
 void PlasmaStore::EraseFromObjectTable(const ObjectID &object_id) {
-  auto &object = store_info_.objects[object_id];
+  auto object = GetObjectTableEntry(&store_info_, object_id);
```

Comment:
Ah nice catch. I assume the symptom of this was we would try to free(nullptr) or similar?

Reply:
It looks like free nullptr is fine (according to the dmalloc doc string, it is no op), but calling get on the default unique ptr seems to have undefined behavior

Reply:
actually calling operator*/-> on default constructed a unique_ptr yields undefined behavior. So calling `PlasmaStore::Free(object->data...)` could potentially lead to free any address.



https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*

> operator* and operator-> provide access to the object owned by *this.
The behavior is undefined if get() == nullptr.